src/frontend/screens/Sync/index.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import {NativeStackNavigationOptions} from '@react-navigation/native-stack';
+
+import {NativeRootNavigationProps} from '../../sharedTypes';
+import {IconButton} from '../../sharedComponents/IconButton';
+import {SettingsIcon} from '../../sharedComponents/icons';
+import {useAllProjects, useProjectSettings} from '../../hooks/server/projects';
+import {useLocalDiscoveryState} from '../../hooks/useLocalDiscoveryState';
+import {CreateOrJoinProjectDisplay} from './CreateOrJoinProjectDisplay';
+import {HeaderTitle} from './HeaderTitle';
+import {NoWifiDisplay} from './NoWifiDisplay';
+import {openWiFiSettings} from '../../lib/linking';
+import {ProjectSyncDisplay} from './ProjectSyncDisplay';
+import {Loading} from '../../sharedComponents/Loading';
+import {useSyncState} from '../../hooks/useSyncState';
+import {useDeviceInfo} from '../../hooks/server/deviceInfo';
+
+export function createNavigationOptions() {
+  return ({
+    navigation,
+  }: NativeRootNavigationProps<'Sync'>): NativeStackNavigationOptions => {
+    return {
+      headerTitleAlign: 'center',
+      headerTitle: () => <HeaderTitle />,
+      headerRight: () => (
+        <IconButton
+          onPress={() => {
+            navigation.navigate('ProjectSettings');
+          }}>
+          <SettingsIcon />
+        </IconButton>
+      ),
+    };
+  };
+}
+
+export const SyncScreen = ({navigation}: NativeRootNavigationProps<'Sync'>) => {
+  // TODO: Is this the right field to use?
+  const wifiStatus = useLocalDiscoveryState(state => state.wifiStatus);
+
+  // TODO: Handle error case
+  const {isLoading, data} = useAllProjects();
+  const syncState = useSyncState();
+  const deviceInfoQuery = useDeviceInfo();
+  const projectSettingsQuery = useProjectSettings();
+
+  if (wifiStatus === 'off') {
+    return (
+      <NoWifiDisplay
+        onOpenSettings={() => {
+          openWiFiSettings().catch(err => {
+            // Should not throw but in case it does, no-op
+            console.warn(err);
+          });
+        }}
+      />
+    );
+  }
+
+  // TODO: Replace with proper check of being a part of a shared project
+  if (data && data.length === 1) {
```

Comment:
highlighting this, even though this needs to be addressed later due to core limitations